src/wallet/wallet2.cpp

```diff
@@ -1616,23 +1617,32 @@ std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& in
   return m_subaddress_labels[index.major][index.minor];
 }
 //----------------------------------------------------------------------------------------------------
-void wallet2::scan_tx(const std::vector<crypto::hash> &txids)
+wallet2::tx_entry_data wallet2::get_sorted_tx_entries(const std::vector<crypto::hash> &txids)
 {
-  // Get the transactions from daemon in batches and add them to a priority queue ordered in chronological order
-  auto cmp_tx_entry = [](const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& l, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& r)
-  { return l.block_height > r.block_height; };
+  tx_entry_data sorted_tx_entries;
+  sorted_tx_entries.tx_entries.reserve(txids.size());
+
+  // don't fetch duplicates
+  std::vector<crypto::hash> txids_no_dups;
+  for (const auto &txid : txids)
+  {
+    if (sorted_tx_entries.tx_hashes.find(txid) == sorted_tx_entries.tx_hashes.end())
+    {
+      sorted_tx_entries.tx_hashes.insert(txid);
+      txids_no_dups.push_back(txid);
+    }
+  }
 
-  std::priority_queue<cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry, std::vector<COMMAND_RPC_GET_TRANSACTIONS::entry>, decltype(cmp_tx_entry)> txq(cmp_tx_entry);
   const size_t SLICE_SIZE =  100; // RESTRICTED_TRANSACTIONS_COUNT as defined in rpc/core_rpc_server.cpp, hardcoded in daemon code
-  for(size_t slice = 0; slice < txids.size(); slice += SLICE_SIZE) {
+  for(size_t slice = 0; slice < txids_no_dups.size(); slice += SLICE_SIZE) {
     cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request req = AUTO_VAL_INIT(req);
     cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response res = AUTO_VAL_INIT(res);
     req.decode_as_json = false;
     req.prune = true;
 
-    size_t ntxes = slice + SLICE_SIZE > txids.size() ? txids.size() - slice : SLICE_SIZE;
+    size_t ntxes = slice + SLICE_SIZE > txids_no_dups.size() ? txids_no_dups.size() - slice : SLICE_SIZE;
     for (size_t i = slice; i < slice + ntxes; ++i)
-     req.txs_hashes.push_back(epee::string_tools::pod_to_hex(txids[i]));
+      req.txs_hashes.push_back(epee::string_tools::pod_to_hex(txids_no_dups[i]));
```

Comment:
Was trying to de-dup in here since I noticed the RPC server doesn't de-dup, updated to just use an `unordered_set` for txids passed to `scan_tx` instead